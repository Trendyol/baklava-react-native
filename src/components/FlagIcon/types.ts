import { flagIconList } from './list';

export type FlagIconNameType = (typeof flagIconList)[number];
