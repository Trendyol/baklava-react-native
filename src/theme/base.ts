export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
