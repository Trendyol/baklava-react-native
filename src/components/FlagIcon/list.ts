export const flagIconList = [
  'czech-republic',
  'germany',
  'greece',
  'qatar',
  'romania',
  'saudi-arabia',
  'slovakia',
  'turkey',
  'united-arab-emirates',
] as const;
