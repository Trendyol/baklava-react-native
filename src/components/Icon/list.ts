export const iconList = [
  'academy',
  'account',
  'add-note',
  'add-photo',
  'add-product',
  'alert',
  'announcement',
  'archive',
  'arrow-down',
  'arrow-left',
  'arrow-right',
  'arrow-up',
  'attach',
  'automatic-pricing',
  'award',
  'back-fill',
  'back',
  'barcode',
  'bill',
  'book',
  'browser',
  'calendar-stroke',
  'calendar',
  'camera',
  'campaign',
  'change-fill',
  'change',
  'chart',
  'chatbot',
  'check-fill',
  'check',
  'clock',
  'close-fill',
  'close',
  'coin',
  'compass',
  'confetti',
  'copy',
  'coupon',
  'delete',
  'dialog',
  'dislike',
  'document-search',
  'document',
  'donation',
  'download',
  'edit',
  'engagement',
  'excel',
  'exit',
  'expand',
  'export',
  'external-link',
  'external-share',
  'eye-off',
  'eye-on',
  'facebook',
  'feedback',
  'filter',
  'finance-accounting',
  'fire',
  'flash-fill',
  'flash',
  'forward',
  'gift',
  'globe',
  'go-forward',
  'graphic',
  'growth',
  'hamburger-menu',
  'hamburger',
  'handshake',
  'happy',
  'heart-line',
  'heart',
  'help',
  'home',
  'image',
  'incognito',
  'influencer',
  'info',
  'label',
  'light-bulb',
  'like',
  'link',
  'listing',
  'live-monitor',
  'live-support',
  'loading',
  'lock',
  'magic-wand',
  'mail-opened',
  'mail',
  'map',
  'market-order-back',
  'market-order-cancel',
  'market-order-change',
  'market-order-check',
  'market',
  'meal-bag',
  'meal-order-back',
  'meal-order-cancel',
  'meal-order-return',
  'message',
  'metric-status-down',
  'metric-status-up',
  'minus-fill',
  'minus',
  'mobile-filled',
  'mobile-settings',
  'mobile-stroke',
  'money-transfer',
  'motorcycle-check',
  'motorcycle',
  'my-offers',
  'normal',
  'notification',
  'order-back',
  'order-box',
  'order-cancel',
  'order-check',
  'order-return',
  'order-tracking',
  'order',
  'paper',
  'pause-fill',
  'pause',
  'people',
  'performance-data',
  'phone-settings',
  'phone',
  'photo',
  'pin',
  'play-fill',
  'play',
  'plus-fill',
  'plus',
  'preview',
  'print',
  'product',
  'profile-check',
  'profile',
  'rating',
  'report',
  'rotate',
  'sad',
  'save',
  'search',
  'seller-ads',
  'send-mail',
  'send',
  'services',
  'settings',
  'share',
  'shopping-bag-back',
  'shopping-bag-cancel',
  'shopping-bag-discount',
  'shopping-bag-return',
  'shopping-bag',
  'sorting',
  'sound-off',
  'sound-on',
  'star',
  'star-fill',
  'stop-fill',
  'stop',
  'store-performance',
  'store',
  'suitcase',
  'support',
  'ticket',
  'truck',
  'turn-back',
  'unlock',
  'upload',
  'user-management',
  'vacation-mode',
  'wallet',
  'warning',
  'web-filled',
  'web-stroke',
  'zoom-in',
  'zoom-out',
] as const;
