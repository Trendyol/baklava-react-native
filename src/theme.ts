import { createTheme } from '@shopify/restyle';

const theme = createTheme({
  colors: {
    white: '#ffffff',
    black: '#000000',
    transparent: 'transparent',
    // Color
    primaryColor: '#F27A1A',
    secondaryColor: '#273142',
    tertiaryColor: '#F1F2F7',
    successColor: '#0BC15C',
    dangerColor: '#FF5043',
    warningColor: '#FFB600',
    alternativeColor: '#5794FF',
    featuredColor: '#8C4EFF',
    // Hover
    primaryHover: '#EF6114',
    secondaryHover: '#0F131A',
    tertiaryHover: '#D5D9E1',
    successHover: '#09A44E',
    dangerHover: '#FF3028',
    warningHover: '#FF9800',
    alternativeHover: '#457EFF',
    featuredHover: '#753EFF',
    // Background
    primaryBackground: '#F1F2F7',
    accentBackground: '#FEF2E8',
    secondaryBackground: '#FFFFFF',
    tertiaryBackground: '#F7F7FA',
    successBackground: '#E7F9EF',
    alertBackground: '#FFEEEC',
    warningBackground: '#FFF8E6',
    infoBackground: '#EEF4FF',
    featuredCBackground: '#F4EDFF',
    // Border Color
    borderColor: '#D5D9E1',
    // Content
    contentPrimary: '#273142',
    contentSecondary: '#6E7787',
    contentTertiary: '#95A1B5',
    contentPassive: '#AFBBCA',
  },
  spacing: {
    none: 0,
    '4xs': 2,
    '3xs': 4,
    '2xs': 8,
    xs: 12,
    s: 14,
    m: 16,
    l: 20,
    xl: 24,
    '2xl': 32,
    '3xl': 40,
    '4xl': 48,
    '5xl': 64,
    '6xl': 80,
  },
  borderRadii: {
    none: 0,
    xs: 2,
    s: 4,
    m: 6,
    l: 8,
    xl: 16,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  fonts: {
    light: 'System',
    regular: 'System',
    medium: 'System',
    semiBold: 'System',
    bold: 'System',
  },
  textVariants: {
    defaults: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      color: 'contentPrimary',
    },
    // Heading
    heading1: {
      fontSize: 32,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 36,
    },
    heading2: {
      fontSize: 28,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 32,
    },
    heading3: {
      fontSize: 24,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 28,
    },
    // Subtitle01
    subtitle01Regular: {
      fontSize: 20,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 24,
    },
    subtitle01Medium: {
      fontSize: 20,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 24,
    },
    subtitle01Semibold: {
      fontSize: 20,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 24,
    },
    subtitle01Bold: {
      fontSize: 20,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 24,
    },
    // Subtitle02
    subtitle02Regular: {
      fontSize: 16,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 20,
    },
    subtitle02Medium: {
      fontSize: 16,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 20,
    },
    subtitle02Semibold: {
      fontSize: 16,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 20,
    },
    subtitle02Bold: {
      fontSize: 16,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 20,
    },
    // Subtitle03
    subtitle03Regular: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    subtitle03Medium: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
    },
    subtitle03Semibold: {
      fontSize: 14,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 16,
    },
    subtitle03Bold: {
      fontSize: 14,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 16,
    },
    // Subtitle04
    subtitle04Regular: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 14,
    },
    subtitle04Medium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 14,
    },
    subtitle04Semibold: {
      fontSize: 12,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 14,
    },
    subtitle04Bold: {
      fontSize: 12,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 14,
    },
    // Body
    bodyText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    bodyUnderline: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyTextLink: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyLongText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    // Caption
    captionText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionMedium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
    },
    captionLongText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionTextLink: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
      textDecorationLine: 'underline',
    },
  },
});

export type Theme = typeof theme;
export default theme;
