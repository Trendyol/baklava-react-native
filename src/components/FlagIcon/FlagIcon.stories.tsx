import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import FlagIcon from './FlagIcon';
import { flagIconList } from './list';
import Box from '../Box/Box';
import Text from '../Text/Text';
import { FlagIconNameType } from './types';

const sizeList = ['2xs', 'xs', 's', 'm', 'l', 'xl'];

const FlagIconMeta: ComponentMeta<typeof FlagIcon> = {
  title: 'FlagIcon',
  component: FlagIcon,
  argTypes: {
    name: {
      options: flagIconList,
      control: { type: 'select' },
    },
    size: {
      options: sizeList,
      control: { type: 'radio' },
    },
  },
  args: {
    name: 'TR',
    size: 'l',
  },
};

export default FlagIconMeta;

type FlagIconStory = ComponentStory<typeof FlagIcon>;

export const Basic: FlagIconStory = args => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Flag Icon
    </Text>
    <Box flexDirection="row" p="2xs" alignItems="center">
      <FlagIcon name={args.name} size={args.size} />
      <Text px="2xs">{args.name}</Text>
    </Box>
  </>
);

export const FlagIconSizes: FlagIconStory = args => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Flag Icon Sizes
    </Text>
    <Box p="2xs" alignSelf="flex-start">
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="xl" />
        <Text px="2xs">xl</Text>
      </Box>
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="l" />
        <Text px="2xs">l</Text>
      </Box>
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="m" />
        <Text px="2xs">m</Text>
      </Box>
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="s" />
        <Text px="2xs">s</Text>
      </Box>
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="xs" />
        <Text px="2xs">xs</Text>
      </Box>
      <Box flexDirection="row" p="2xs" alignItems="center">
        <FlagIcon name={args.name} size="2xs" />
        <Text px="2xs">2xs</Text>
      </Box>
    </Box>
  </>
);

export const FlagIconList: FlagIconStory = () => (
  <>
    <Text p="2xs" variant="subtitle01Bold">
      Flag Icon List
    </Text>
    <Box p="2xs" alignSelf="flex-start">
      {flagIconList.map((iconName: FlagIconNameType) => (
        <Box key={iconName} flexDirection="row" p="2xs" alignItems="center">
          <FlagIcon name={iconName} />
          <Text px="2xs">{iconName}</Text>
        </Box>
      ))}
    </Box>
  </>
);
