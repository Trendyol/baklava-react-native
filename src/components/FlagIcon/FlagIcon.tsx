import React from 'react';
import { useTheme, VariantProps } from '@ergenekonyigit/restyle';
import * as icons from '../../icons';
import { Theme } from '../../theme';
import { FlagIconNameType } from './types';
import { toPascalCase } from './utils';
import { SvgProps } from 'react-native-svg';

const FlagIcon = ({
  name,
  size = 'l',
  testID = 'flagIcon',
  ...rest
}: SvgProps & {
  name: FlagIconNameType;
  size?: VariantProps<Theme, 'iconSizeVariants'>['variant'];
}) => {
  const theme = useTheme<Theme>();

  if (!name) {
    return null;
  }

  const iconSize = theme.iconSizeVariants[size];

  const TheIcon = icons[toPascalCase(name) as keyof typeof icons];

  return (
    <TheIcon
      {...rest}
      title={name}
      width={iconSize}
      height={iconSize}
      testID={testID}
      accessibilityLabel={testID}
    />
  );
};

export default FlagIcon;
