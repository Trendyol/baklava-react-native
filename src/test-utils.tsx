import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
