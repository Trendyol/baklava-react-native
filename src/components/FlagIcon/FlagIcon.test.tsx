import React from 'react';
import theme from '../../theme';
import { render } from '../../test-utils';
import FlagIcon from './FlagIcon';
import { flagIconList } from './list';

describe('FlagIcon', () => {
  beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation(() => ({}));
  });

  test('should render icon correctly', () => {
    // when
    const { toJSON } = render(<FlagIcon name="turkey" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render default size correctly', () => {
    // when
    const { getByTestId } = render(
      <FlagIcon testID="flagIcon" name="turkey" />,
    );
    const iconComponent = getByTestId('flagIcon');

    // then
    expect(iconComponent.props.width).toBe(theme.iconSizeVariants.l);
    expect(iconComponent.props.height).toBe(theme.iconSizeVariants.l);
  });

  test('should render given size correctly', () => {
    // when
    const { getByTestId } = render(
      <FlagIcon testID="flagIcon" name="turkey" size="2xs" />,
    );
    const iconComponent = getByTestId('flagIcon');

    // then
    expect(iconComponent.props.width).toBe(theme.iconSizeVariants['2xs']);
    expect(iconComponent.props.height).toBe(theme.iconSizeVariants['2xs']);
  });

  test('should render iconList correctly', () => {
    // when
    const { getByTestId } = render(
      <FlagIcon testID="flagIcon" name={flagIconList[6]} />,
    );
    const iconComponent = getByTestId('flagIcon');

    // then
    expect(iconComponent).toBeTruthy();
  });

  test('should not render icon component without name', () => {
    // when
    //@ts-ignore
    const { queryByTestId } = render(<FlagIcon testID="flagIcon" />);
    const iconComponent = queryByTestId('flagIcon');

    // then
    expect(iconComponent).toBeNull();
  });
});
