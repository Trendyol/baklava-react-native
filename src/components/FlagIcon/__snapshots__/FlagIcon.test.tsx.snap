// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagIcon should render icon correctly 1`] = `
<RNSVGSvgView
  accessibilityLabel="flagIcon"
  align="xMidYMid"
  bbHeight={24}
  bbWidth={24}
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  testID="flagIcon"
  title="turkey"
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGGroup
      clipPath="a"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGPath
        d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12Z"
        fill={
          {
            "payload": 4292345895,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="m15.165 9.98.985 1.356 1.595-.517-.987 1.356.985 1.357-1.594-.52-.986 1.356v-1.676l-1.593-.519 1.594-.517.001-1.676Z"
        fill={
          {
            "payload": 4293980400,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M12.478 15.565a3.391 3.391 0 1 1 1.613-6.375 4.174 4.174 0 1 0 0 5.967c-.48.26-1.029.408-1.613.408Z"
        fill={
          {
            "payload": 4293980400,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
    <RNSVGDefs>
      <RNSVGClipPath
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        name="a"
      >
        <RNSVGPath
          d="M0 0h24v24H0z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGClipPath>
    </RNSVGDefs>
  </RNSVGGroup>
</RNSVGSvgView>
`;
